scripts/setup.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+set -euo pipefail
+
+setup_gpg() {
+    echo "Setting up GPG with agent for automated commit signing..."
+
+    # Validate required files/vars exist
+    if [[ ! -f "$GPG_KEY_FILE" ]]; then
+        echo "Error: No GPG key file found at $GPG_KEY_FILE" >&2
+        exit 1
+    fi
+    
+    if [[ -z "${GPG_SECRET:-}" ]]; then
+        echo "Error: GPG_SECRET environment variable not set" >&2
+        exit 1
+    fi
+
+    import_key
+    setup_agent
+    setup_cron_refresh
+    
+    echo "GPG setup with agent caching complete"
+}
+
+import_key() {
+    echo "Importing GPG key..."
+    
+    # Import the key
+    if ! echo "$GPG_SECRET" | gpg --batch --yes --passphrase-fd 0 --pinentry-mode loopback --import "$GPG_KEY_FILE"; then
+        echo "Error: Failed to import GPG key" >&2
+        exit 1
+    fi
+    
+    # Get key info
+    local fingerprint=$(gpg --list-secret-keys --with-colons | grep '^fpr:' | head -n 1 | cut -d: -f10)
```

Comment:
@cywang117 recommend running these shell scripts through shellcheck to catch some obvious gotchas.